app/Main.hs

```diff
@@ -149,6 +154,28 @@ main = do
                               return ()
          ActionCompile -> compileProg config
 
+-- create an output relation for each input relation
+mirrorInputRelations :: DatalogProgram -> String -> DatalogProgram
```

Comment:
This function should probably be in `DatalogProgram.hs` and should be called `progMirrorInputRelations`
Likewise, `makeOutputRelations` should be `DatalogProgram.progOutputInternalRelations` or something similar.